dataactbroker/handlers/accountHandler.py

```diff
@@ -494,42 +529,46 @@ def update_user(self, system_email):
 
         """
         sess = GlobalDB.db().session
-        request_dict = RequestDictionary(self.request)
-
-        # throw an exception if nothing is provided in the request
-        if not request_dict.exists("uid") or not (request_dict.exists("status") or request_dict.exists("permissions") or
-                    request_dict.exists("is_active")):
-            # missing required fields, return 400
-            exc = ResponseException("Request body must include uid and at least one of the following: status, permissions, is_active",
-                                    StatusCode.CLIENT_ERROR)
-            return JsonResponse.error(exc, exc.status)
+        request_dict = RequestDictionary.derive(self.request)
 
         try:
+            editable_fields = ('status', 'permissions', 'is_active')
+            has_editable = any(key in request_dict for key in editable_fields)
+            if 'uid' not in request_dict or not has_editable:
+                # missing required fields, return 400
+                raise ResponseException(
+                    "Request body must include uid and at least one of the "
+                    "following: status, permissions, is_active",
+                    StatusCode.CLIENT_ERROR
+                )
             # Find user that matches specified uid
-            user = sess.query(User).filter(User.user_id == int(request_dict.getValue("uid"))).one()
-        except NoResultFound:
-            exc = ResponseException("No users with that uid", StatusCode.CLIENT_ERROR)
+            user = sess.query(User).filter_by(
+                user_id=int(request_dict['uid'])).one_or_none()
+            if user is None:
```

Comment:
I don't think it's always the right answer; in these cases, I more wanted to avoid nested `try`s :)